src/main/java/com/box/sdk/BinaryBodyUtils.java

```diff
@@ -71,4 +93,31 @@ static void writeStreamTo(InputStream input, OutputStream output) {
             }
         }
     }
+
+    static void writeStreamTo(InputStream input, OutputStream output, long expectedLength) {
+        long totalBytesRead = 0;
+        if (expectedLength < 0) {
+            throw new RuntimeException("No Data bytes in stream");
```

Comment:
Content-Length header should never be negative. I guess this check is not needed